lib/rbi/parser.rb

```diff
@@ -156,32 +156,25 @@ class TreeBuilder < Visitor
       sig { returns(Tree) }
       attr_reader :tree
 
-      sig { returns(T.nilable(YARP::Node)) }
+      sig { returns(T.nilable(Prism::Node)) }
       attr_reader :last_node
 
-      sig { params(source: String, comments: T::Array[YARP::Comment], file: String).void }
+      sig { params(source: String, comments: T::Array[Prism::Comment], file: String).void }
       def initialize(source, comments:, file:)
         super(source, file: file)
 
-        @comments_by_line = T.let(comments.to_h { |c| [c.location.start_line, c] }, T::Hash[Integer, YARP::Comment])
+        @comments_by_line = T.let(comments.to_h { |c| [c.location.start_line, c] }, T::Hash[Integer, Prism::Comment])
         @tree = T.let(Tree.new, Tree)
 
         @scopes_stack = T.let([@tree], T::Array[Tree])
-        @last_node = T.let(nil, T.nilable(YARP::Node))
+        @last_node = T.let(nil, T.nilable(Prism::Node))
         @last_sigs = T.let([], T::Array[RBI::Sig])
         @last_sigs_comments = T.let([], T::Array[Comment])
       end
 
-      sig { override.params(node: T.nilable(YARP::Node)).void }
-      def visit(node)
-        return unless node
-
-        @last_node = node
-        super
-      end
-
-      sig { override.params(node: YARP::ClassNode).void }
+      sig { override.params(node: Prism::ClassNode).void }
       def visit_class_node(node)
+        @last_node = node
```

Comment:
Since we're now only setting `last_node` for the node we have an explicit visit on, should we set it to `nil` before leaving the node to avoid errors pointing to the wrong node?

Reply:
Good point! Done